### PR TITLE
codeintel: Clean up transport packages

### PR DESCRIPTION
**Pull request by efritz**
More untangling of graphql packages.

## Test plan

Ran locally.

### CODE REVIEW

File: enterprise/internal/codeintel/autoindexing/transport/graphql/root_resolver_inference.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+		resolvers = append(resolvers, resolver)
+	}
+
+	return resolvers, nil
+}
+
+//
+//
+
+type autoIndexJobDescriptionResolver struct {
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/autoindexing/transport/graphql/root_resolver_inference.go:123]: Would you move this to its own file? It feels out of place. Not blocking of course, just wondering.

File: enterprise/internal/codeintel/uploads/transport/graphql/iface.go [hunk 2]
```diff
@@ -8,48 +8,40 @@ import (
...
-	InferIndexConfiguration(ctx context.Context, repositoryID int, commit string, localOverrideScript string, bypassLimit bool) (_ *config.IndexConfiguration, hints []config.IndexJobHint, err error)
-	QueueIndexes(ctx context.Context, repositoryID int, rev, configuration string, force bool, bypassLimit bool) (_ []types.Index, err error)
-
-	GetSupportedByCtags(ctx context.Context, filepath string, repoName api.RepoName) (bool, string, error)
-	GetLanguagesRequestedBy(ctx context.Context, userID int) (_ []string, err error)
-	SetRequestLanguageSupport(ctx context.Context, userID int, language string) (err error)
-}
-
 type UploadsService interface {
 	sharedresolvers.UploadsService
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/uploads/transport/graphql/iface.go:15]: you could inline the two methods from `sharedresolvers.UploadsService`

```
GetIndexesByIDs(ctx context.Context, ids ...int) (_ []types.Index, err error)
GetUploadsByIDs(ctx context.Context, ids ...int) (_ []types.Upload, err error)
```

save yourself the import _shrug_

File: enterprise/internal/codeintel/uploads/transport/graphql/root_resolver_coverage.go [hunk 3]
```diff
@@ -20,19 +24,123 @@ import (
...
+
+	indexJobs, err := r.autoindexSvc.InferIndexJobsFromRepositoryStructure(ctx, id, commit, "", false)
+	if err != nil {
+		if !autoindexing.IsLimitError(err) {
+			return nil, err
+		}
+
+		limitErr = errors.Append(limitErr, err)
+	}
+	// indexJobHints, err := r.autoindexSvc.InferIndexJobHintsFromRepositoryStructure(ctx, repoID, commit)
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/uploads/transport/graphql/root_resolver_coverage.go:90]: artifact from developing? Should it be uncommented?

Comment (efritz) [reply enterprise/internal/codeintel/uploads/transport/graphql/root_resolver_coverage.go:90]: These were already commented as a "kind a working but too low signal/noise". These were just moved from a deleted file.

File: enterprise/internal/codeintel/uploads/transport/graphql/root_resolver_index_queries.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+	resolverstubs "github.com/sourcegraph/sourcegraph/internal/codeintel/resolvers"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+const DefaultPageSize = 50
+
+func (r *rootResolver) PreciseIndexes(ctx context.Context, args *resolverstubs.PreciseIndexesQueryArgs) (_ resolverstubs.PreciseIndexConnectionResolver, err error) {
+	ctx, errTracer, endObservation := r.operations.preciseIndexes.WithErrors(ctx, &err, observation.Args{LogFields: []log.Field{
+		// log.String("uploadID", string(id)),
```
Comment (Numbers88s) [inline enterprise/internal/codeintel/uploads/transport/graphql/root_resolver_index_queries.go:24]: is this also an artifact from development?

Comment (efritz) [reply enterprise/internal/codeintel/uploads/transport/graphql/root_resolver_index_queries.go:24]: Also moved from before but we could fill it out.

**Review by Numbers88s — APPROVED** [review]
Dude. It looks great! 🤩

File: enterprise/internal/codeintel/autoindexing/transport/graphql/root_resolver_inference.go (hunk 1 above)
Comment (efritz) [reply enterprise/internal/codeintel/autoindexing/transport/graphql/root_resolver_inference.go:123]: I'd like to keep the files highly coupled in the graphql packages while we rename the API surface so we know what can actually cross-call to what. We can relax this in a bit though and give resolvers their own breathing room.

Related, I also have another idea in which we can use struct _fields_ with this GraphQL library for things that don't need to re-fetch data.

File: enterprise/internal/codeintel/uploads/transport/graphql/iface.go (hunk 2 above)
Comment (efritz) [reply enterprise/internal/codeintel/uploads/transport/graphql/iface.go:15]: Right now all of the find-refs for each of the iface methods only occur in the package their declared in. If we inline them we lose that, and I'm currently using it as a dead code detector check. We can also relax this once we're done moving things around.

**Pull request merged**